js/custom-post-parents-admin.js

```diff
@@ -1,61 +1,45 @@
+
 ;(function ($) {
     $(function () {
         var $ul = $('.custom-post-parents-selectedlist');
-        var $adders = $('.custom-post-parents-selectlist-add');
-
-        $ul.on('click', '.ntdelbutton', function () {
+        $ul.on('click','.ntdelbutton',function(){
+            var id_option=$(this).attr("name");
+            var section=$(this).attr("data-section");
+            $("#cpt-f-"+section+" >option[id='"+id_option+"']").show();
```

Comment:
Llevas razón ya que la comprobación del id es innecesaria porque es único. He omitido la parte del selector a partir del +section, quedando de la siguiente manera:

        $ul.on('click','.ntdelbutton',function(){
            var current_item=$(this);
            var section=current_item.data("section");
            $("#cpt-f-"+section).show();
            current_item.parents('span').remove();
        });